weblab/repocache/models.py

```diff
@@ -124,6 +124,202 @@ class Meta:
         unique_together = ['entity', 'tag']
 
 
+####################################################################################################
+#
+# Mixin classes defining the common cache structure
+#
+
+class CachedEntityMixin(models.Model):
```

Comment:
I'm going to need some pointers around these mixins please so I can appreciate what is going on. In the mean time I will keep digging

Reply:
Ok., I think I get it now. Its just code common to the three child classes. sorry being thick

Reply:
That's right!